apps/site/pages/en/learn/test-runner/mocking.md

```diff
@@ -0,0 +1,264 @@
+  });
+});
+```
+
+In the above, the first and second cases (the `it`s) can sabotage each other because they are run concurrently and mutate the same store (a race condition): "save"'s insertion can cause the otherwise valid "read"'s test to fail its assertion on items found (and "read"'s can do the same thing to "save"'s).
+
+## What to mock
+
+### Modules + units
+
+```mjs
+import assert from 'node:assert/strict';
+import { before, describe, it, mock } from 'node:test';
+
+
+describe('foo', { concurrency: true }, () => {
+  let barMock = mock.fn();
+  let foo;
+
+  before(async () => {
+    const barNamedExports = await import('./bar.mjs')
+      // discard the original default export
+      .then(({ default, ...rest }) => rest);
+
+    // It's usually not necessary to manually call restore() after each
+    // nor reset() after all (node does this automatically).
+    mock.module('./bar.mjs', {
+      defaultExport: barMock
+      // Keep the other exports that you don't want to mock.
+      namedExports: barNamedExports,
+    });
+
+    // This MUST be a dynamic import because that is the only way to ensure the
+    // import starts after the mock has been set up.
+    // There is a far more technical explanation,
+    // but just trust that this is logically necessary.
+    ({ foo } = await import('./foo.mjs'));
+  });
+
+  it('should do the thing', () => {
+    barMock.mockImplementationOnce(function bar_mock() {/* … */});
+
+    assert.equal(foo(), 42);
+  });
+});
+```
+
+### Services
+
+A little-known fact: node has a builtin way to mock `fetch`. [`undici`](https://github.com/nodejs/undici) is the Node.js implementation of fetch. You do not have to install it—it's shipped with node by default.
+
+```mjs displayName="endpoints.spec.mjs"
+import assert from 'node:assert/strict';
+import { beforeEach, describe, it } from 'node:test';
+import { MockAgent, setGlobalDispatcher } from 'undici';
+
+import endpoints from './endpoints.mjs';
+
+describe('endpoints', { concurrency: true }, () => {
+  let agent;
+  beforeEach(() => {
+    agent = new MockAgent();
+    setGlobalDispatcher(agent);
+  });
+
+  it('should retrieve data', async () => {
+    const endpoint = 'foo';
+    const code = 200;
+    const data = {
+      key: 'good',
+      val: 'item',
+    };
+
+    agent
+      .get('example.com')
+      .intercept({
+        path: endpoint,
+        method: 'GET',
+      })
+      .reply(code, data);
+
+    assert.deepEqual(await endpoints.get(endpoint), {
+      code,
+      data,
+    });
+  });
+
+  it('should save data', async () => {
+    const endpoint = 'foo/1';
+    const code = 201;
+    const data = {
+      key: 'good',
+      val: 'item',
+    };
+
+    agent
+      .get('example.com')
+      .intercept({
+        path: endpoint,
+        method: 'PUT',
+      })
+      .reply(code, data);
+
+    assert.deepEqual(await endpoints.save(endpoint), {
+      code,
+      data,
+    });
+  });
+});
+```
+
+### Time
```

Comment:
May be add mention of TZ. This can affect tests as part of a unit test of front-end code.